activity_calendar/models.py

```diff
@@ -134,44 +134,48 @@ def get_activitymoments_between(self, start_date, end_date):
         # Any occurrence before (start_date - duration) can never partially take place inside the
         #   specified time period, so there's no need to look back even further.
         activity_duration = self.duration
-        occurrences = self.get_occurrences_starting_between(start_date - activity_duration, end_date)
+        recurrency_occurences = self.get_occurrences_starting_between(start_date - activity_duration, end_date)
 
         # Note that DST-offsets have already been handled earlier
         #   in self.get_occurrences_between -> util.dst_aware_to_dst_ignore
 
-        # The above comment isn't entirely True as there are also activity_moments we need to consider,
-        #   which we handle here
+        # Get the correct activitymoment instances from the database
+        activity_moments_between_query = Q(recurrence_id__gte=(start_date - activity_duration)) &\
+                                         Q(recurrence_id__lte=end_date)
+        # Get a list of all activitymoments that due to shift are either
+        # incorrectly included (surplus) or wrongly excluded (extra)
         surplus_moments, extra_moments = self._get_queries_for_alt_start_time_activity_moments(start_date, end_date)
 
         # Filter out all activitymoments with an alt start time outside the bounds
-        surplus_moments = self.activitymoment_set.filter(surplus_moments).values_list('recurrence_id', flat=True)
-        occurrences = filter(lambda occ: occ not in surplus_moments, occurrences)
+        surplus_moments_queryset = self.activitymoment_set.filter(surplus_moments).values_list('recurrence_id', flat=True)
+        recurrency_occurences = filter(lambda occ: occ not in surplus_moments_queryset, recurrency_occurences)
 
         # Evaluate the iterable: we want to use it more than once below
```

Comment:
I don't think this is the case anymore now that line 156 was changed to also include activitymoments outside of the recurrence scheme.